app/src/main/res/layout/item_intro_2.xml

```diff
@@ -8,8 +8,10 @@
 
   <ImageView
     android:id="@+id/airplane"
-    android:layout_width="0dp"
-    android:layout_height="wrap_content"
+    android:layout_width="wrap_content"
+    android:layout_height="0dp"
+    app:layout_constraintHeight_max="121dp"
```

Comment:
Why 121? I would prefer if ratios/biases were used to give a uniform experience across screen sizes

Reply:
121 is the actual size of the drawable. Earlier the height was set as wrap_content but that is not possible with app:layout_constraintHeight_max so I hard-coded the size of the image. The height and width of the kiwix logo in item_intro_1 had also been hard-coded so I assumed it was okay to do.

Reply:
> item_intro_1 had also been hard-coded so I assumed it was okay to do.

very understandable mistake.

The more fundamental issue to me seems that the text's top wasn't constrained to the image's bottom, these could even be a horizontal chain if they wanted to be? Just thinking out loud. Point is contraintlayout is not being utilised to the full.

I'd prefer if we never relied on hardcoded values, if you are willing to take another shot at this I'd enjoy that. Otherwise you did fix a bug and I don't exactly feel terrible or anything about accepting it, it just could be improved.

Reply:
Since the image's bottom is already constrained to the top of the text, it isn't letting me constraint in the other direction at the same time.
In any case either the image will have to shrink in horizontal mode like it does in my fix, or we will have to use a ScrollView to ensure a fixed size but then we wont be able to align the text to be at the bottom of the screen. That would make the text appear in the middle of the screen on taller phones.